pkg/shared/clients/redis/redis_reader.go

```diff
@@ -122,17 +126,28 @@ func (br *RedisStreamsRead) processReadError(xstreams []redis.XStream, messages
 // send array of 1 element.
 func (br *RedisStreamsRead) Ack(_ context.Context, offsets []isb.Offset) []error {
 	errs := make([]error, len(offsets))
+	// if we were to have n messages produced from 1 incoming message, we could have
+	// the same offset more than once: just in case, we can deduplicate
+	dedupOffsets := make(map[string]struct{}) // essentially a Set
 	strOffsets := []string{}
 	for _, o := range offsets {
-		strOffsets = append(strOffsets, o.String())
+		_, found := dedupOffsets[o.String()]
+		if !found {
+			dedupOffsets[o.String()] = struct{}{}
+			strOffsets = append(strOffsets, o.String())
+		}
 	}
 	if err := br.Client.XAck(RedisContext, br.Stream, br.Group, strOffsets...).Err(); err != nil {
 		for i := 0; i < len(offsets); i++ {
-			errs[i] = err
+			errs[i] = err // 'errs' is indexed the same as 'offsets'
+		}
+		if br.Metrics.AckErrorsAdd != nil {
+			br.Metrics.AckErrorsAdd(len(strOffsets))
+		}
+	} else {
+		if br.Metrics.AcksAdd != nil {
+			br.Metrics.AcksAdd(len(strOffsets))
 		}
```

Comment:
this dedup logic isn't really being used at the moment since we're publishing 1 message for every incoming message, but if in the future we were to configure an option to publish multiple messages in the case of multiple key/value pairs then we would need this